### PR TITLE
Use comms-template library

### PR DESCRIPTION
**Pull request by laurence-bird**
Updating the composer to use the comms-templates library to retrieve templates from s3

### CODE REVIEW

File: src/main/scala/com/ovoenergy/comms/aws/TemplateContextFactory.scala [hunk 1]
```diff
@@ -18,8 +16,6 @@ object S3ClientFactory {
           new ProfileCredentialsProvider()
         )
     }
-    val underlying: AmazonS3Client =
-      new AmazonS3Client(awsCredentials).withRegion(Regions.fromName(region))
-    new AmazonS3ClientWrapper(underlying)
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/aws/TemplateContextFactory.scala]: The `S3Client` trait and its `AmazonS3ClientWrapper` implementation can be deleted

File: src/main/scala/com/ovoenergy/comms/repo/S3TemplateRepo.scala [hunk 2]
```diff
@@ -1,85 +1,33 @@
...
-          EmailTemplate(
-            subject = sub,
-            htmlBody = html,
-            textBody = textBody,
-            sender = sender,
-            htmlFragments = htmlFragments,
-            textFragments = textFragments
-          )
+  def getEmailTemplate(commManifest: CommManifest) = ReaderT[ErrorOr, TemplatesContext, EmailTemplate[Id]] { context =>
+    TemplatesRepo
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/repo/S3TemplateRepo.scala]: I know we flipped-flopped a few times on this, but having a long one-liner including a match looks weird to me. I would write this as

```scala
val emailTemplate = TemplatesRepo.getTemplate(context, commManifest).map(_.email)
emailTemplate match {
  case Invalid(errs) => Left( ... )
  case Valid(None) => Left( ...)
  case Valid(Some(result)) => Right(result)
}
```

File: src/main/scala/com/ovoenergy/comms/email/EmailTemplate.scala [hunk 3]
```diff
@@ -1,10 +0,0 @@
-package com.ovoenergy.comms.email
-
-import com.ovoenergy.comms.Mustache
-
-case class EmailTemplate(subject: Mustache,
```
Comment (cb372) [inline src/main/scala/com/ovoenergy/comms/email/EmailTemplate.scala]: `Mustache.scala` can also be deleted

**Comment by ovo-comms-circleci** [comment]
This PR was successfully deployed to production by [CircleCI build 130](https://circleci.com/gh/ovotech/comms-composer/130) at 2017-01-31 16:24 GMT.

**Pull request merged**